tests/core/services/document_manipulation/test_document_manipulation.py

```diff
@@ -4,205 +4,70 @@
-
-    @patch(
-        "app.core.services.document_manipulation.document_manipulation._parse_if_valid_pdf",
-        return_value=encrypted_pdf_mock,
-    )
-    @patch(
-        "app.core.services.document_manipulation.document_manipulation._get_sanitize_offset",
-        return_value=0,
-    )
-    def test_split_encrypted_pdf(self, mock_sanitize_offset, mock_parse_pdf):
-        result = document_manipulation.split_pdf(io.BytesIO(b"ciao"), 2, 5)
-        self.assertEqual(result.read(), b"ciao")
-        self.assertEqual(mock_parse_pdf.call_count, 1)
-        self.assertEqual(mock_sanitize_offset.call_count, 1)
+    # Then
+    assert result is None
 
 
-def test_given_pdf_without_extra_headers_sanitize_offset_should_return_0(expect):
+def test_split_pdf_invalid_pdf(expect):
     # Given
+    empty_pdf = b"%PDF-1.7"
     buff_argument = io.BytesIO()
-    pdf_content: List[bytes] = [b"%PDF-1.5", b"pdf_content1", b"pdf_content2"]
-    expect(document_manipulation, times=1)._read_buffer_line_by_line(
+    expect(document_manipulation, times=1)._parse_if_valid_pdf(
         buff_argument
-    ).thenReturn(pdf_content)
+    ).thenReturn(None)
+    expect(document_manipulation, times=1)._write_pdf_to_buffer(None).thenReturn(
+        empty_pdf
+    )
 
     # When
-    result = document_manipulation._get_sanitize_offset(buff_argument)
+    result = document_manipulation.split_pdf(buff_argument, 0, 1)
 
     # Then
-    assert result == 0
+    assert result is empty_pdf
 
 
-def test_given_pdf_with_one_extra_headers_sanitize_offset_should_return_len_first_string(
-    expect,
-):
+def test_split_pdf_valid_pdf_no_pages_to_split(expect):
     # Given
     buff_argument = io.BytesIO()
-    pdf_content: List[bytes] = [b"extra-header" b"%PDF-1.5", b"pdf_content1"]
-    expect(document_manipulation, times=1)._read_buffer_line_by_line(
+    expect(document_manipulation, times=1)._parse_if_valid_pdf(
         buff_argument
-    ).thenReturn(pdf_content)
+    ).thenReturn(True)
 
     # When
-    result = document_manipulation._get_sanitize_offset(buff_argument)
+    result = document_manipulation.split_pdf(buff_argument, 1, 0)
 
     # Then
-    assert result == 12
+    assert result.read() is b""
 
 
 @pytest.mark.parametrize(
-    "in_out_tuple",
+    "in_out_extensions_tuple",
     [
-        ([b"x%PDF-1.5"], 1),
-        ([b"xxxx%PDF-1.5"], 4),
-        ([b"xxxxxxxxxxx%PDF-1.5"], 11),
-        ([b"x%PDF-1.5", b"pdf-content"], 1),
-        ([b"pdf-content", b"x%PDF-1.5"], 12),
-        ([b"xxxx%PDF-1.5", b"pdf-content"], 4),
-        ([b"pdf-content", b"xxxx%PDF-1.5"], 15),
-        ([b"xxxxxxxxxxx%PDF-1.5", b"pdf-content"], 11),
-        ([b"pdf-content", b"xxxxxxxxxxx%PDF-1.5"], 22),
+        ("jpeg", "png"),
+        ("JPEG", "png"),
+        ("png", "png"),
+        ("PNG", "png"),
+        ("other_ext", "other_ext"),
     ],
 )
-def test_given_pdf_with_extra_headers_sanitize_offset_should_return_correct_size(
```

Comment:
Are these parameterized tests removed because the validity of a pdf is done by pypdfium2 and for this reason we trust it?

Reply:
Yes!